geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CreateJndiBindingFunction.java

```diff
@@ -54,19 +49,17 @@ public CliFunctionResult executeFunction(FunctionContext<Object[]> context)
     try {
       JNDIInvoker.mapDatasource(getParamsAsMap(configuration),
           convert(configuration.getConfigProperties()));
-    } catch (DataSourceCreateException ex) {
+    } catch (DataSourceCreateException dsce) {
       if (logger.isErrorEnabled()) {
-        logger.error("create " + TYPE_NAME + " failed", ex.getWrappedException());
+        logger.error("create " + TYPE_NAME + " failed", dsce.getWrappedException());
       }
-      throw ex;
-    } catch (NamingException ex) {
+      return new CliFunctionResult(context.getMemberName(), StatusState.ERROR, dsce.getMessage());
+    } catch (Exception ex) {
       if (logger.isErrorEnabled()) {
-        logger.error("create " + TYPE_NAME + " failed", ex);
+        logger.error("create " + TYPE_NAME + " failed", ex.getMessage());
```

Comment:
It is good to log the exception `ex` itself, instead of `ex.getMessage()`. So that the stack trace can be logged. It will help analyze the exception.

Reply:
Done